options.go

```diff
@@ -34,6 +34,13 @@ import (
 // An Option is an option for configuring a Dialer.
 type Option func(d *dialerConfig)
 
+// NetResolver groups the methods on net.Resolver that are used by the DNS
```

Comment:
If we're adding something to the public API for tests only, we should change our approach (maybe by patching an unexported field).

If on the other hand we think this is a useful interface for callers, we should remove "This allows the default net.Resolver instance to be overridden from tests" and elaborate on why a person might want to provide their own interface.

Reply:
Also, nit if we're keeping this interface let's move it down to just above where it's used.

Reply:
Moved to just above the `WithResolver()` function. 

I think this makes sense as a public API. I can imagine a customer keeping their DNS entries in a separate, private DNS server. Then they would need to configure the connector to use that DNS server.